python-checks/src/test/java/org/sonar/python/checks/quickfix/PythonQuickFixVerifier.java

```diff
@@ -96,6 +97,8 @@ public static void verify(Function<String, PythonVisitorContext> createVisitorCo
       .as("The code with the quickfix applied is not the expected result.\n" +
         "\"Applied QuickFixes are:\n%s\nExpected result:\n%s", appliedQuickFix, Arrays.asList(codesFixed))
       .isEqualTo(Arrays.asList(codesFixed));
+
+    assertThatCode( () -> PythonParser.createIPythonParser().parse(String.join("\n", codesFixed))).doesNotThrowAnyException(); // We assume that any IPython code is valid Python code
```

Comment:
On another note, I do have the feeling that we could refactor a bit the behavior for the `createIPythonVisitorContext` and `createPythonVisitorContext` methods so that we propagate the parser they use, which would mean we can use the correct one here as well.

I don't have a very strong opinion that we should do this, although I think it may avoid mistakes where we quick fix Python code into IPython code - however unlikely that is to happen.